PR template: Request manual repro instructions (#35102)
hashicorp/terraform: .github/pull_request_template.md

<!--

> Describe in detail the changes you are proposing, and the rationale. If practical, describe how to manually test the change; this isn't always possible, but it makes PRs easier and faster to review.

Reviewer: ```suggestion
Your pull request should be accompanied by a GitHub issue that describes in detail the changes you are proposing, and the rationale.
```

See the contributing guide:

https://github.com/hashicorp/terraform/blob/main/.github/CONTRIBUTING.md

-->

> 

Reviewer: ```suggestion
Fixes #

<!-- Link all GitHub issues fixed by this PR, and add references to prior related PRs. -->
```

Bring this up to the top since it's the most important thing.

> ## Reason for change
> 
> ## Old behavior
> 
> ## New behavior

Reviewer: I have two related concerns:

1. Adding all these fields makes the PR template pretty long and could be a potential barrier to someone submitting a PR, especially if it's something simple.
2. Our contributing guide already sets the standard that [changes should be proposed in a GitHub issue first](https://github.com/hashicorp/terraform/blob/main/.github/CONTRIBUTING.md#proposing-a-change), so that would be the right place to describe this information, and not in the PR itself.

Thankfully both of these concerns can be addressed by a single solution: emphasize that a GitHub issue is the place to include this info, which simplifies the PR template quite a bit.

```suggestion
```

## How to verify

> 

Reviewer: ```suggestion

<!-- If practical, describe how to manually test the change; this isn't always possible, but it makes PRs easier and faster to review. -->

```

### Example config

> 

Reviewer: Can we give an example here in a `<!-- comment -->`?

### Commands

> 

Reviewer: Can we give an example here in a `<!-- comment -->`?

> <!--
> 
> Link all GitHub issues fixed by this PR, and add references to prior

Reviewer: ```suggestion
```

Remove this plus lines 26 through 31. (GitHub UI is not letting me make that suggestion for whatever reason.)

Describe in detail the changes you are proposing, and the rationale.

> Also, if it's practical, please provide an example config and commands for manually testing this change with Terraform CLI. (This isn't always possible, but it can enable better and faster reviews for your PR.)

Reviewer 2: Perhaps we could add some headings so the contributor can just fill them in?

e.g. `#Old behaviour`, `#New behaviour`, `#How to test ##Example config ##Commands`

Author: @kmoe That's a good idea! 

I updated the PR to give that a try; it might have too many headers, but let's see what folks think.
